pkg/cluster/clusterserviceprincipal.go

```diff
@@ -0,0 +1,136 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/Azure/ARO-RP/pkg/util/arm"
+	"github.com/Azure/ARO-RP/pkg/util/rbac"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+func (m *manager) createOrUpdateClusterServicePrincipalRBAC(ctx context.Context) error {
+	resourceGroup := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+
+	clusterSPObjectID, err := m.aad.GetServicePrincipalID(ctx)
+	if err != nil {
+		return err
+	}
+
+	roleAssignments, err := m.roleassignments.ListForResourceGroup(ctx, resourceGroup, fmt.Sprintf("principalId eq '%s'", clusterSPObjectID))
```

Comment:
I wonder whether we are going to need to have robust logic that validates all role assignments and adds/removes if necessary.  I suspect that this would be the place for it.